Update DAC doc for experiment event data (#38490)
DataDog/documentation: content/en/account_management/rbac/data_access.md

### Agent Observability

> * **Supported telemetry**: Agent Observability traces are supported. Experiment event data (spans and evaluation metrics) for experiments belonging to a project is also restricted by `ml_app`-keyed Restricted Datasets; only the event data is restricted—experiment list views and metadata are not. Datasets, annotation queues, and managed prompts are not supported.

Reviewer: ```suggestion
* **Supported telemetry**: Agent Observability traces are supported. Experiment event data (spans and evaluation metrics) for experiments in a project is also restricted by `ml_app`-keyed Restricted Datasets. Only the event data is restricted—experiment list views and metadata are not. Datasets, annotation queues, and managed prompts are not supported.
```
